cm/master.py

```diff
@@ -259,11 +259,11 @@ def load_legacy_snapshot_data(self):
         if s3_conn and misc.get_file_from_bucket(s3_conn, self.app.config['bucket_default'],
            'snaps.yaml', snaps_file, validate=validate):
             pass
-        elif misc.get_file_from_public_bucket(self.app.config, self.app.config['bucket_default'],
-                                              'snaps.yaml', snaps_file):
+        elif misc.get_file_from_public_location(self.app.config, 'snaps.yaml', snaps_file):
             log.warn("Couldn't get snaps.yaml from bucket: {0}. However, managed "
-                     "to retrieve it from public S3 bucket {0} instead."
-                     .format(self.app.config['bucket_default']))
+                     "to retrieve it from public location '{1}' instead.".format(
+                     self.app.config['bucket_default'],
+                     self.app.config.get('default_bucket_url') or self.app.config['bucket_default']))
```

Comment:
Right. I'd say using the same approach as below (ie, `config.get('bucket_default', None)`) would be preferred because there's no guarantee the filed `bucket_default` is actually defined in the `config` dict?